src/main/scala/intellij/haskell/spellchecker/CabalSpellcheckingStrategy.scala

```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.spellchecker
+
+import com.intellij.psi.PsiElement
+import com.intellij.spellchecker.tokenizer.{SpellcheckingStrategy, Tokenizer}
+import intellij.haskell.cabal.CabalLanguage
+
+/**
+  * Provide spellchecker support for Cabal sources.
+  */
+class CabalSpellcheckingStrategy extends SpellcheckingStrategy {
```

Comment:
The document is [here](https://www.jetbrains.com/help/idea/2016.2/spellchecking.html), there is a built-in dictionary inside intellij, and [here](https://github.com/rikvdkleij/intellij-haskell/blob/master/src/main/resources/dictionary/haskell.dic) is a explcit dictionary for haskell words which we want to be accepted by the typo checker.